src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -118,36 +129,51 @@ class TooltipRenderedOnPageBody extends React.PureComponent {
             this.props.maxWidth,
             this.state.tooltipWidth,
             this.state.tooltipHeight,
-            this.state.tooltipTextWidth,
+            this.state.tooltipContentWidth,
             this.props.shiftHorizontal,
             this.props.shiftVertical,
         );
-        return ReactDOM.createPortal(
-            <Animated.View
-                onLayout={this.measureTooltip}
-                style={[tooltipWrapperStyle, animationStyle]}
-            >
+
+        let content;
+        const contentRef = (ref) => {
+            // Once the content for the tooltip first renders, update the width of the tooltip dynamically to fit the width of the content.
+            // Note that we can't have this code in componentDidMount because the ref for the content won't be set until after the first render
+            if (this.contentRef) {
+                return;
+            }
+
+            this.contentRef = ref;
+            this.updateTooltipContentWidth();
+        };
+
+        if (this.props.renderTooltipContent) {
```

Comment:
NAB: Move `let content;` to be just before `if (this.props.renderTooltipContent) {`